### PR TITLE
Filtering Port

### PR DESCRIPTION
**Pull request by ciliakas**
A simplified version of our previous filtering and totals calculating system, and should be able to do almost  everything our previous system did.
Entry.cshtml.cs is an example of how to use it

### CODE REVIEW

File: src/Pages/Entry.cshtml.cs [hunk 1]
```diff
@@ -32,7 +35,8 @@ public void OnGet()
             IGoal goal2 = new Goal();
             //IEntry goal3 = new Goal(); -- thats illegla
 
-            ViewData["EntryList"] = entry1.ToString();
+            //This should be ill eagle, don't know why I didn't think of this before
+            ViewData["EntryList"] = dataManager.Income.EntryList.GetBy(Importance.Low).GetBy(DateTime.Today).GetSum();
```
Comment (Stokalas) [inline src/Pages/Entry.cshtml.cs:39]: This seems very neat

File: src/DataManagement/Entries/EntryList.cs [hunk 2]
```diff
@@ -8,12 +10,58 @@ public class EntryList : List<IEntry>, IEntryList
...
+            EntryType = entryType;
+            if (entryList is null) return;
+            AddRange(entryList);
+        }
+        
+        public IEntryList GetBy(Importance importance)
+        {
+            return new EntryList(EntryType, this.Where(x => x.Importance == (int)importance).ToList());
+        }
+
```
Comment (Stokalas) [inline src/DataManagement/Entries/EntryList.cs:29]: Are all these methods avaible for Income/Expenses or for the Goals as well? Not necessarily that we would use it for goals, just interesting.

**Review by Stokalas — APPROVED** [review]
Since this one is a lot shorter, not much to say. I really like how you simplified all those filtering/total methods into so few. Really well done. And as always I've left few questions.

_Approving since no flaws found_

File: src/DataManagement/Entries/EntryList.cs (hunk 2 above)
Comment (ciliakas) [reply src/DataManagement/Entries/EntryList.cs:29]: I've just added to Incomes and Expenses for now at least

**Review by Stokalas — COMMENTED** [review]
Just a question, so if I want to get entries only from this month I should use ?

**Comment by ciliakas** [comment]
> 
> 
> Just a question, so if I want to get entries only from this month I should use ?

GetBy(DateTime.Today)

**Pull request merged**